tests/middleware/test_base.py

```diff
@@ -1153,3 +1147,52 @@ async def send(message: Message) -> None:
         {"type": "http.response.body", "body": b"good!", "more_body": True},
         {"type": "http.response.body", "body": b"", "more_body": False},
     ]
+
+
+# XXX: Do we even need `test_anyio_streams_cleanup_*`?
+#  1. Warning is triggered in other tests for unhandled exceptions. These tests
+#  are a bit special because they trigger `gc.collect()` so, warnings trigger
+#  within a test and not when GC runs.
+#  2. Tests will become evergreen if `anyio` decides to stop emitting warnings
+#  on non-closed streams deletion: https://github.com/agronholm/anyio/blob/4.6.2.post1/src/anyio/streams/memory.py#L183
```

Comment:
If you think this is not necessary, you are free to remove them.

I'm already happy with the removal of the warning in the `pyproject.toml`, and the current tests not failing.

Reply:
I think we can remove them.

Reply:
Removed